src/Utilities/RemediationPairs.js

```diff
@@ -0,0 +1,131 @@
+import chunk from 'lodash/chunk';
+
+const REQUEST_CHUNK_SIZE = 1000;
+const BATCH_REQUEST_SIZE = 5;
+const REQUEST_INTERVAL = 15000; //15 econds. AKAMAI allowed life for 5 API calls
+
+const fetchDataCallback  = (endpoint) => (input) =>
+    fetch(`/api/patch/v1/views${endpoint}`, {
+        method: 'POST',
+        credentials: 'include',
+        headers: {
+            Accept: 'application/json',
+            'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(input)
+    }).then(result =>
+        result.status === 200
+            ? result.json()
+            : Promise.reject(result.statusText)
+    );
+
+export const transformPairs = (input) => {
+    return {
+        issues: Object.keys(input?.data || {}).map(advisory => {
+            return {
+                id: `patch-advisory:${advisory}`,
+                description: advisory,
+                systems: input.data[advisory]
+            };
+        }
+        )
+    };
+};
+
+const transformSystemsPairs = (input) => {
+    const pairs = [];
+    Object.entries(input.data).map(
+        ([systemID, advisories]) => {
+            advisories.map(advisory => {
+                const pairID = `patch-advisory:${advisory}`;
+                const index = pairs.findIndex(pair => pair.id === pairID);
+                if (index >= 0) {
+                    pairs[index].systems.push(systemID);
+                } else {
+                    pairs.push(
+                        {
+                            id: pairID,
+                            description: advisory,
+                            systems: [systemID]
+                        }
+                    );
+                }
```

Comment:
A nit-pick: this seems to be inefficient, especially when you search for indices with `findIndex`. Probably, you could use `reduce` here together with `pairs` set as an object, not array? This will let you easier work and update the dictionary (object). In the end, you can transform it back to array...

Reply:
Actually, I started pairs as an object initially but later had to convert it to array type. Let me remember why I switched to Array.